### PR TITLE
PC-235

### PR DESCRIPTION
**Pull request by eyalvo-matrix**

### CODE REVIEW

**Comment by eyalvo-matrix** [comment]
This is draft only no function explanation yet

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/StrategyElement/Encounter/FhirEncounterMapping.php [hunk 1]
```diff
@@ -21,9 +20,6 @@
 use OpenEMR\FHIR\R4\FHIRElement\FHIREncounterStatus;
 use OpenEMR\FHIR\R4\FHIRResource\FHIREncounter\FHIREncounterParticipant;
 
-use  FhirAPI\FhirRestApiBuilder\Parts\Strategy\Traits\ConversionsTrait;
```
Comment (eyalvo-matrix) [inline modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/StrategyElement/Encounter/FhirEncounterMapping.php:24]: use is in base mapping

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    public function validate($validator, $data, $mainTable = null)
+    {
+        $this->valueSet = new ValueSet(array(
+            'paramsFromUrl' => array(),
+            'paramsFromBody' => array(),
+            'container' => $this->container
+        ));
+
+        switch ($validator['validation']) {
+            case 'blockedStatusFinished':
```
Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php:34]: change to dynamic function with $status parameter as we talked in the meeting.

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+     *
+     * @param $validator array
+     * @param $data array
+     * @param $mainTable string
+     *
+     * @return bool
+     */
+    public function validate($validator, $data, $mainTable = null)
+    {
+        $this->valueSet = new ValueSet(array(
```
Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php:27]: better to init the new ValueSet in the checkIfInList method, with singlton logic (prevent unnecessary initializes)

Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php:27]: ok

File: modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php (hunk 2 above)
Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/FhirRestApiBuilder/Parts/Strategy/Traits/FHIRElementValidation.php:34]: ok

File: modules/FhirAPI/sql/install.sql [hunk 4]
```diff
@@ -340,3 +340,76 @@ INSERT INTO `fhir_value_sets` (`id`, `title`) VALUES
...
+('Condition', 'outcome', 'WRITE', 'valueset', 'condition_statuses', 'DB', 1),
+('Condition', 'type', 'WRITE', 'required', '', 'DB', 1),
+('Condition', 'date', 'WRITE', 'required', '', 'DB', 1),
+('Condition', 'diagnosis', 'WRITE', 'required', '', 'DB', 1),
+('Condition', 'pid', 'WRITE', 'ifExist', 'patient_data', 'DB', 1),
+('Condition', 'user', 'WRITE', 'ifExist', 'users', 'DB', 1),
+('MedicationStatement', 'outcome', 'WRITE', 'valueset', 'medication_statement_statuses', 'DB', 1),
+('MedicationStatement', 'diagnosis', 'WRITE', 'required', '', 'DB', 1),
+('MedicationStatement', 'pid', 'WRITE', 'ifExist', 'patient_data', 'DB', 1),
+('Observation', 'activity', 'WRITE', 'valueset', 'observation_statuses', 'DB', 1),
```
Comment (matrix-amiel) [inline modules/FhirAPI/sql/install.sql:394]: missing blockStatusIfValue for status final of observation

File: modules/FhirAPI/sql/install.sql [hunk 5]
```diff
@@ -340,3 +340,76 @@ INSERT INTO `fhir_value_sets` (`id`, `title`) VALUES
...
+('MedicationRequest', 'interval', 'WRITE', 'valueset', 'drug_interval', 'DB', 1),
+('MedicationRequest', 'active', 'WRITE', 'valueset', 'medicationrequest_status', 'DB', 1),
+('MedicationRequest', 'patient_id', 'WRITE', 'ifExist', 'patient_data', 'DB', 1),
+('MedicationRequest', 'user', 'WRITE', 'ifExist', 'users', 'DB', 1),
+('MedicationRequest', 'encounter', 'WRITE', 'required', '', 'DB', 1),
+('MedicationRequest', 'instruction_code', 'WRITE', 'valueset', 'tests_and_treatments', 'DB', 1),
+('MedicationRequest', 'status', 'WRITE', 'valueset', 'servicerequest_statuses', 'DB', 1),
+('MedicationRequest', 'intent', 'WRITE', 'valuesetNotRequired', 'servicerequest_intent', 'DB', 1),
+('MedicationRequest', 'patient', 'WRITE', 'ifExist', 'patient_data', 'DB', 1),
+('ServiceRequest', 'encounter', 'WRITE', 'required', '', 'DB', 1),
```
Comment (matrix-amiel) [inline modules/FhirAPI/sql/install.sql:411]: missing blockStatusIfValue for status completed of service request

**Pull request merged**